src/screens/AddressList.js

```diff
@@ -0,0 +1,213 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import { t } from 'ttag'
+import HathorPaginate from '../components/HathorPaginate';
+import HathorAlert from '../components/HathorAlert';
+import { EXPLORER_BASE_URL, WALLET_HISTORY_COUNT } from '../constants';
+import helpers from '../utils/helpers';
+import path from 'path';
+
+import hathorLib from '@hathor/wallet-lib';
+
+/**
+ * Screen that has a list of addresses of the wallet
+ *
+ * @memberof Screens
+ */
+class AddressList extends React.Component {
+  /**
+   * addresses {Array} All wallet addresses data {'address', 'index', 'numberOfTransactions'}
+   * filteredAddresses {Array} addresses state after search
+   * page: {Number} Current page of the list
+   * totalPages: {Number} Total number of pages of the list
+   * filtered: {Boolean} If the list is filtered
+   */
+  state = {
+    addresses: [],
+    filteredAddresses: [],
+    page: 1,
+    totalPages: 0,
+    filtered: false,
+  }
+
+  componentDidMount = () => {
+    const walletData = hathorLib.wallet.getWalletData();
+    const addressKeys = walletData.keys;
+    const addresses = [];
+    for (const key in addressKeys) {
+      const addressData = {
+        address: key,
+        index: addressKeys[key].index,
+        numberOfTransactions: this.getNumberOfTransactions(key, walletData.historyTransactions)
+      };
+
+      addresses.push(addressData);
+    }
+
+    this.setState({ addresses, filteredAddresses: addresses, totalPages: this.getTotalPages(addresses) });
+  }
+
+  /**
+   * Return total number of pages of the list
+   *
+   * @param {Array} array Array of addresses of the list
+   *
+   * @return {Number} Total number of pages of the list
+   */
+  getTotalPages = (array) => {
+    return Math.ceil(array.length / WALLET_HISTORY_COUNT);
+  }
+
+  /**
+   * Return total number of transactions using an address
+   *
+   * @param {String} address Address to get the total number of transactions
+   * @param {Object} historyTransactions History of transactions from the wallet data storage
+   *
+   * @return {Number} Total number of transactions using this address
+   */
+  getNumberOfTransactions = (address, historyTransactions) => {
```

Comment:
This algorithm seems really slow for big wallets because we are iterating all transactions multiple times, one for each address. I think it can receive the `addressKeys` and iterate only once storing the counters in a dictionary. Then, it returns the dictionary. Something like this:

```
counter = {};
for (const key in addressKeys) {
  counter[key] = 0;
}
```

Then, instead of `if (el.decoded.address === address)`, it just have to check if the key exists `if (el.decoded.address in counter)`.

Reply:
Updated the code. Can you see if it's good?